bridge-fungible-token/src/utils.sw

```diff
@@ -19,14 +28,61 @@ const GTF_INPUT_MESSAGE_DATA = 0x11E;
 const GTF_INPUT_MESSAGE_SENDER = 0x115;
 const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
 
+fn bn_mul(bn: U256, d: u64) -> (b256, u64) {
+    disable_panic_on_overflow();
```

Comment:
The reason I was storing the flags and then restoring was because I didn't want to disturb whatever setting the developer had up to the point. At the end of this function, you re-enable the panic even if the developer had turned it off before which could cause problems for them.

It's really a hack that it's disabled in the assembly at all. Given the u64 input restriction, there isn't a way to actually cause overflow in our calculation, we just want to use the overflow register to do it more efficiently.

Reply:
> Given the u64 input restriction, there isn't a way to actually cause overflow in our calculation

Wouldn't using this function with inputs like this overflow:
`bn_mult(U256::max(), 2u64)` ?

Reply:
>The reason I was storing the flags and then restoring was because I didn't want to disturb whatever setting the developer had up to the point.

This makes sense for the general-purpose use case (i.e: when this is ported to the stdlib). 
There's no chance of that happening in this specific contract as this is the only place we touch the flags.
I feel like for a contract that will likely be copied and deployed many times it's probably best to be as explicit as possible (less asm is better) as showcase best practices for general usage.
A dev who want to modify the contract and manually configure flags ought to be capable of  modifying this as needed, but most never will I would think.